Umweltzone/src/main/java/de/avpptr/umweltzone/zones/ZonesFragment.kt

```diff
@@ -23,23 +23,24 @@ import de.avpptr.umweltzone.BuildConfig
 import de.avpptr.umweltzone.R
 import de.avpptr.umweltzone.Umweltzone
 import de.avpptr.umweltzone.base.BaseFragment
+import de.avpptr.umweltzone.databinding.FragmentZonesBinding
 import de.avpptr.umweltzone.models.AdministrativeZone
 import de.avpptr.umweltzone.utils.ContentProvider
 import de.avpptr.umweltzone.utils.IntentHelper
 import de.avpptr.umweltzone.zones.dataconverters.toZoneViewModels
 import de.avpptr.umweltzone.zones.viewmodels.ZoneViewModel
-import kotlinx.android.synthetic.main.fragment_zones.*
 
 class ZonesFragment : BaseFragment() {
 
     override fun getLayoutResource(): Int = R.layout.fragment_zones
 
-    override fun onActivityCreated(savedInstanceState: Bundle?) {
-        super.onActivityCreated(savedInstanceState)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
         val context = requireContext()
         administrativeZones = ContentProvider.getAdministrativeZones(context)
         val zoneViewModels = administrativeZones.toZoneViewModels(context)
-        administrativeZonesView.adapter = ZonesAdapter(zoneViewModels, ::onItemClick)
+        val binding = FragmentZonesBinding.bind(view)
```

Comment:
What do you think about nulling the binding in `onDestroyView()` as pointed out in the [documentation](https://developer.android.com/topic/libraries/view-binding#fragments)?

Reply:
I wondered about that, but I interpreted it as the binding needs to be set to null if it's a class member. Since it holds a reference to the view, it would prevent the view from being garbage collected. In this case, the binding is only used in the local method, so when it goes out of scope it should be available for garbage collection. Granted, I haven't found any examples to support this simplification. The ones I found keep the binding as a member.

Reply:
The generated class files can be found in *Umweltzone/build/generated/data_binding_base_class_source_out/debug/out/de/avpptr/umweltzone/databinding/*, e.g. `FragmentZonesBinding.java`. Maybe this helps.

Reply:
Here, the `binding` should be garbage collected because the `ZonesFragment` is tied to its corresponding `ZonesActivity`. It is not put to the fragments backstack when another screen is opened on top.